app/src/main/java/com/goayo/debtify/viewmodel/AddDebtViewModel.java

```diff
@@ -74,18 +79,27 @@ public void setSelectedBorrowersData(Set<IUserData> borrowers) {
      * @param borrowers   the borrower/borrowers
      * @param amount      the total amount of the debt
      * @param description a brief description of the debt
+     * @param isNoSplit   How the debt will be split
      * @throws Exception to be specified later
      */
     public void createDebt(String groupID,
                            Set<IUserData> lender,
                            Set<IUserData> borrowers,
                            BigDecimal amount,
-                           String description) throws Exception {
+                           String description,
+                           boolean isNoSplit) throws Exception {
         // TODO refactor all strings to IUserData
+        IDebtSplitStrategy splitStrategy;
+        if (isNoSplit) {
+            splitStrategy = new NoSplitStrategy();
+        } else {
+            splitStrategy = new EvenSplitStrategy();
+        }
```

Comment:
One thing we could do is create a ISplitStrategyFactory to remove the ViewModels dependency on the specific implementation of each splitStrategy and instead move the dependency to just the ISplitStrategy-interface.
ISplitStrategyFactory have methods "createNoSplitStrategy() : ISplitStrategy" and "createEvenSplitStrategy() : ISplitStrategy" which returns the specific ISplitStrategy implementation but it is wrapped in the generic ISplitStrategy interface.
This way ViewModels no longer create the object but just receive it wrapped from factory.

However we might be treading on "too much abstraction"-territory with this implementation. But it's worth a thought!